contracts/MyMultiSig.cdc

```diff
@@ -263,6 +152,89 @@ pub contract MyMultiSigV2 {
             destroy action
         }
 
+        pub fun createMultiSign(action: {Action}): UInt64 {
+            let newAction <- create MultiSignAction(signers: self.signers.keys, action: action)
+            let uuid = newAction.uuid
+            self.actions[newAction.uuid] <-! newAction
+            emit ActionCreated(uuid: uuid, intent: action.intent)
+            return uuid
+        }
+
+        pub fun signerApproveAction(actionUUID: UInt64, messageSignaturePayload: MessageSignaturePayload) {
+            pre {
+                self.signers[messageSignaturePayload.signingAddr] == true:
+                    "This address is not allowed to sign for this."
+                self.borrowAction(actionUUID:actionUUID).accountsVerified[messageSignaturePayload.signingAddr] != true:
+                    "This address has already signed."
+            }
+            let action = self.borrowAction(actionUUID:actionUUID)
+
+            // Validate Message
```

Comment:
We are validating message signature for both approving action and rejecting action, is it possible to abstract this process into a private function and have the two public ones call it?

Reply:
we are already sort of doing this -- both `signerApprove` and `signerReject` functions have:

```
// Validate Message
assert(
       MyMultiSigV2.approveOrRejectActionMessageIsValid(action: action, messageSignaturePayload: messageSignaturePayload),
        message: "Signed message is invalid"
)
```

Reply:
Sorry about the incomplete message, what I mean is abstracting the process of `validateMessage` and `validateSignature` for both approving action and rejecting action

Reply:
Gotcha!!! Its a good suggestion!  The reason I haven't done that is because I've been anticipating that we might eventually change the message format for these two functions, so the validations would be slightly different.  I've kept them the same for now just to make it easy to implement and test, but I think we need to have a broader conversation on what the contents of the message actually should be for each function to make it as secure as possible.

To summarize -- having identical message for approving & rejecting has some security issues, and I plan on changing it. :)